rust/ql/lib/codeql/rust/dataflow/internal/DataFlowImpl.qll

```diff
@@ -260,11 +260,13 @@ module Node {
     /** Gets the node before the state update. */
     Node getPreUpdateNode() { result = TExprNode(n) }
 
+    override ExprCfgNode asExpr() { result = n }
```

Comment:
We don't want this; this predicate should only hold for `TExprNode`.

Reply:
Yes, I realised that 😅